pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -274,50 +277,72 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 
+	if klusterlet.DeletionTimestamp.IsZero() {
+		if !hasFinalezer(klusterlet, klusterletFinalizer) {
+			return n.addFinalezer(ctx, klusterlet, klusterletFinalizer)
+		}
+	}
+
 	// TODO: remove this when detached mode is not used in klusterlet
 	if config.InstallMode == operatorapiv1.InstallModeDetached {
 		config.InstallMode = operatorapiv1.InstallModeHosted
 	}
 
 	if config.InstallMode == operatorapiv1.InstallModeHosted {
+		var newStatus *operatorapiv1.KlusterletStatus
+		var updateErr error
 		managedClusterClients, err = n.buildManagedClusterClientsHostedMode(ctx, n.kubeClient, config.AgentNamespace, config.ExternalManagedKubeConfigSecret)
 		if err != nil {
-			_, _, _ = helpers.UpdateKlusterletStatus(ctx, n.klusterletClient, klusterletName, helpers.UpdateKlusterletConditionFn(metav1.Condition{
+			newStatus, _, updateErr = helpers.UpdateKlusterletStatus(ctx, n.klusterletClient, klusterletName, helpers.UpdateKlusterletConditionFn(metav1.Condition{
```

Comment:
I wonder whether we just set the condition, and update at last? Also explain why we do not return err here.

Reply:
it is not easy to determine where is the "last", it may return in different places.
And notes added to explain why we do not return here